### PR TITLE
Docs: Add PostsPerPage XML doc

### PR DESCRIPTION
**Pull request by GaryJones**
See #1722.

### CODE REVIEW

**Comment by GaryJones** [comment]
I adapted the wording from the sniff, but adjusted it to remove the reference to high post counts coming from the use of `posts_per_page = -1`. See https://github.com/WordPress/WordPress-Coding-Standards/issues/1638 for context around this.

**Comment by jrfnl** [comment]
@GaryJones Just wondering if you'll have a chance to finish this off in the near future.
I'd like to release WPCS 2.2.0 soon and it would be great if this PR could be included.

**Comment by GaryJones** [comment]
I'm working on this today / tomorrow.

File: WordPress/Docs/WP/PostsPerPageStandard.xml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+<documentation title="Flag returning high or infinite posts_per_page.">
```
[the remark this answers is not in the document]
Comment (GaryJones) [reply WordPress/Docs/WP/PostsPerPageStandard.xml]: Done.

File: WordPress/Docs/WP/PostsPerPageStandard.xml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    <standard>
+    <![CDATA[
+Using posts_per_page (or numberposts) with the value set to an unreasonably high number or setting nopaging to true opens up the potential for scaling issues if the query ends up querying thousands of posts.
```
[the remark this answers is not in the document]
Comment (GaryJones) [reply WordPress/Docs/WP/PostsPerPageStandard.xml]: All bullet points done.

I've mentioned the default, and that it's configurable, in the description. I've kept the `(default 100)` in each description, as that is a key part of understanding why the numbers chosen are valid or not valid (i.e. what the limit is).

File: WordPress/Docs/WP/PostsPerPageStandard.xml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        <code title="Valid: posts_per_page is not over limit (default 100).">
+        <![CDATA[
+$args = array(
+    'posts_per_page' => <em>-1</em>,
+    'posts_per_page' => <em>100</em>,
+    'posts_per_page' => <em>'10'</em>,
+);
+
+$query_args['posts_per_page'] = <em>100</em>;
+$query_args['posts_per_page'] = <em>'10'</em>;
```
[the remark this answers is not in the document]
Comment (GaryJones) [reply WordPress/Docs/WP/PostsPerPageStandard.xml]: Done.

File: WordPress/Docs/WP/PostsPerPageStandard.xml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        ]]>
+        </code>
+        <code title="Invalid: posts_per_page is over limit (default 100).">
+        <![CDATA[
+$args = array(
+    'posts_per_page' => <em>999</em>,
+);
+
```
[the remark this answers is not in the document]
Comment (GaryJones) [reply WordPress/Docs/WP/PostsPerPageStandard.xml]: Done.

File: WordPress/Docs/WP/PostsPerPageStandard.xml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+_query_posts( 'nopaging=1&posts_per_page=999' );
```
[the remark this answers is not in the document]
Comment (GaryJones) [reply WordPress/Docs/WP/PostsPerPageStandard.xml]: Added.

File: WordPress/Docs/WP/PostsPerPageStandard.xml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+You should always fetch the lowest number possible that still gives you the number of results you find acceptable.
+    ]]>
+    </standard>
+    <code_comparison>
+        <code title="Valid: posts_per_page is not over limit (default 100).">
+        <![CDATA[
+$args = array(
+    'posts_per_page' => <em>-1</em>,
+    'posts_per_page' => <em>100</em>,
+    'posts_per_page' => <em>'10'</em>,
```
[the remark this answers is not in the document]
Comment (GaryJones) [reply WordPress/Docs/WP/PostsPerPageStandard.xml]: I've split them up into separate arrays.

**Comment by GaryJones** [comment]
> I'm also wondering whether the code comparison _split on the key used_ is actually needed or that these could just be combined into one which shows both keys.
> 
> Or maybe better yet: whether there should be separate code comparisons for each way to set the keys. That would mean three code comparisons:

I've kept them split as is for now, but I'd be OK with merging them into a single valid + invalid comparisons.

**Comment by jrfnl** [comment]
@GaryJones Looks like something went wrong with your cross-branch merging as this PR now seems to contains all the changes since WCEU.... Could you please try to fix this ? If you need help getting the branch cleaned up, please let me know.

File: WordPress/Docs/WP/PostsPerPageStandard.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jrfnl) [reply WordPress/Docs/WP/PostsPerPageStandard.xml]: @GaryJones Looking at the docs now, I understand the choice for having the default in the `Valid/Invalid` descriptions, but in that case, the "_The default limit of 100 is configurable._" in the standard description seems superfluous. Or maybe it should just be moved to the end of the description as it now breaks up the explanation.

**Comment by jrfnl** [comment]
@GaryJones I've had a look at just the doc file, presuming that's the only chance in this PR.

Other than my reply about the placement of the info around the default limit, it all looks good to me.

Once the branch is fixed, I'm happy to approve and merge this.

File: WordPress/Docs/WP/PostsPerPageStandard.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (GaryJones) [reply WordPress/Docs/WP/PostsPerPageStandard.xml]: I'll remove it.

**Comment by GaryJones** [comment]
> Looks like something went wrong with your cross-branch merging as this PR now seems to contains all the changes since WCEU

@jrfnl Rebased into a single commit, and fixed the branch issue - all clean again :-)

**Pull request merged**